lib/certmanager/certmanager.go

```
@@ -44,49 +48,60 @@ func LoadSigningCredentials(ctx context.Context, certConfig *CertificateConfig)
 }
 
 func getCertificate(certPath string) (*x509.Certificate, error) {
+	ctx := context.Background()
 
 	certBytes, err := os.ReadFile(certPath)
 	if err != nil {
+		log.Logger.Error(ctx, "failed to read certificate file", err, nil)
```

no need to log **errors** at every level, getCertificate is returning an error which is already being logged in LoadSigningCredentials.
So lib should simply return the errors, and they will be logged at the top level in the service.
Lib should have only info, debug and warning logs for monitoring